skbio/io/format/genbank.py

```diff
@@ -0,0 +1,683 @@
+           'left_partial_': False,
+           'right_partial_': False}
+    items = re.split('[(),]+', loc_str)
+    operators = ['join', 'complement']
+    if 'complement' in items:
+        res['rc_'] = True
+    for i in items:
+        i = i.strip()
+        if i in operators or not i:
+            continue
+        elif '..' in i:  # span
+            beg, end = i.split('..')
+            if beg.startswith('<'):
+                beg = beg[1:]
+                res['left_partial_'] = True
+            if end.startswith('>'):
+                end = end[1:]
+                res['right_partial_'] = True
+            beg = int(beg)
+            end = int(end)
+            index = range(beg-1, end)
+        elif i.isdigit():  # single base
+            index = int(i) - 1
+        else:
+            raise GenbankFormatError(
+                'Could not parse location string:\n%s' %
+                loc_str)
+        pmd[index] = True
+
+    return res, pd.Series(pmd)
+
+
+def _parse_origin(lines):
+    '''Parse the ORIGIN section for sequence.
+    '''
+    sequence = []
+    for line in lines:
+        if line.startswith('ORIGIN'):
+            continue
+        # remove the number at the beg of each line
+        items = line.split()
+        sequence.append(''.join(items[1:]))
+    return ''.join(sequence)
+
+
+def _serialize_origin(seq, indent=9):
+    '''Serialize seq to ORIGIN.
+
+    Parameters
+    ----------
+    seq : str
+    '''
+    n = 1
+    line_size = 60
+    frag_size = 10
+    for i in range(0, len(seq), line_size):
+        line = seq[i:i+line_size]
+        s = '{n:>{indent}} {s}\n'.format(
+            n=n, indent=indent, s=chunk_str(line, frag_size, ' '))
+        if n == 1:
+            s = 'ORIGIN\n' + s
+        n = n + line_size
+        yield s
+
+
+def _parse_section_default(
+        lines, label_delimitor=None, join_delimitor=' ', return_label=False):
+    '''Parse sections in default way.
+
+    Do 2 things:
+        1. split first line with label_delimitor for label
+        2. join all the lines into one str with join_delimitor.
+    '''
+    data = []
+    first = True
+    label = None
+    for line in lines:
+        if first:
+            try:
+                items = line.split(label_delimitor, 1)
+            except:
+                GenbankFormatError('Could not split the line:\n%s', line)
+            if len(items) == 2:
+                label, section = items
+            else:
+                label = items[0]
+                section = ""
+            data.append(section)
+            first = False
+        else:
+            data.append(line)
+    data = join_delimitor.join(i.strip() for i in data)
+    if return_label:
+        return label, data
+    else:
+        return data
+
+
+def _serialize_section_default(header, obj, indent=12):
+    return '{header:<{indent}}{obj}\n'.format(
+        header=header, obj=obj, indent=indent)
+
+
+def yield_section(is_another_section, **kwargs):
```

Comment:
this could be a generic useful function and i am thinking about move it to
_base.py. what do you think?

On Tue, Aug 18, 2015 at 9:23 AM, Jai Ram Rideout <notifications@github.com>
wrote:

> In skbio/io/format/genbank.py
> <https://github.com/biocore/scikit-bio/pull/1023#discussion_r37319666>:
>
> > +            first = False
> > +        else:
> > +            data.append(line)
> > +    data = join_delimitor.join(i.strip() for i in data)
> > +    if return_label:
> > +        return label, data
> > +    else:
> > +        return data
> > +
> > +
> > +def _serialize_section_default(header, obj, indent=12):
> > +    return '{header:<{indent}}{obj}\n'.format(
> > +        header=header, obj=obj, indent=indent)
> > +
> > +
> > +def yield_section(is_another_section, **kwargs):
>
> Can you rename this to _yield_section? All of the other functions in this
> module start with an underscore.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/1023/files#r37319666>.
>